cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -490,6 +504,43 @@ private void closeScanner() {
     }
   }
 
+  /**
+   * Filters elements matching an index  {@link ScannerIterator}.
+   */
+  @VisibleForTesting
+  static final class FilterByIndexIterator extends AbstractCloseableIterator<StructuredRow> {
+    private final LimitIterator limitIterator;
+    private final Field<?> filterIndex;
+
+    FilterByIndexIterator(LimitIterator limitIterator, Field<?> filterIndex) {
+      this.limitIterator = limitIterator;
+      this.filterIndex = filterIndex;
+    }
+
+    @Override
+    protected StructuredRow computeNext() {
+      // Postfiltering on scanned rows by index
+      while (limitIterator.hasNext()) {
+        StructuredRow row = limitIterator.next();
+        if (row.getString(filterIndex.getName()).equals(filterIndex.getValue())) {
```

Comment:
need to handle the case if row.getString() returns a null

Reply:
Perhaps use `Objects.equals(row.getString(filterIndex.getName()), filterIndex.getValue())`?